### PR TITLE
[#6086] Teams is adding support for suggested actions in 1-1 chats

### LINKED ISSUE #6086: Teams is adding support for suggested actions in 1-1 chats
[Prompts](https://github.com/microsoft/botbuilder-dotnet/blob/main/libraries/Microsoft.Bot.Builder.Dialogs/Prompts/Prompt.cs#L309) and [IntputDialogs](https://github.com/microsoft/botbuilder-dotnet/blob/main/libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Input/InputDialog.cs#L393) call `ChoiceFactory.ForChannel` by default if a `ListStyle` is not provided. 

`ForChannel` will construct a card using `SuggestedActions` if possible. This process should now also check if the conversation is Teams 1:1, in which case `SuggestedActions` ARE supported. (note: only Ring0 as of 1.3.2021)

https://github.com/microsoft/botbuilder-dotnet/issues/6065#issuecomment-1004334158

### PR DESCRIPTION
**Pull request by ceciliaavila**
Fixes #6086

## Description
This PR enables Suggested Actions for _MSTeams_ in Dialogs Prompts and Adaptive Dialogs Inputs.

## Specific Changes
  - Updated **_ChoiceInput_** , **_ConfirmInput_** , **_ChoicePrompt_**, and **_ConfirmPrompt_** classes to pass to the  _AppendChoices_ method the _conversationType_ and the list of recipient's ids, necessary to send suggested actions in the _MSTeams_ channel.
  - Updated _AppendChoices_ method in **_InputDialog_** and **_Prompt_** overloading it to receive the new parameters _conversationType_ and _toList_ and pass them to the _ForChannel_ method.
  - Updated _SupportsSuggestedActions_ method in **_Channel_** class to include the MSTeams channel in the switch case.
  - Overloaded the _ForChannel_ method in **_ChoiceFactory_** and the _SuggestedActions_ method of the **_MessageFactory_** to add the new parameters _conversationType_ and _toList_.
  - Added unit tests in **_ChoiceFactoryTests_** and **_ChoiceChannelTests_** test classes.
  - Updated **_MessageFactoryTests_** to disambiguate the calls to _SuggestedActions_.

## Testing
These images show how MSTeams displays the choices in a group chat (hero card) and in a personal chat (suggested actions).
image

### CODE REVIEW

**Comment by BruceHaley** [comment]
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.Dialogs.dll**

**Comment by limamicro** [comment]
Hi @ceciliaavila, can you give me the sample code of the bot reply logic, as well as the manifest file?
I try to reproduce your changes to verify the behavior, but I could not see the expected behavior as you attached in Testing result.
Thanks!

**Comment by limamicro** [comment]
@ceciliaavila regarding the changes, right now suggested actions is only available 1:1 scope. All suggested actions sent to group chat and channels should be dropped instead of supporting that. Can you help me to understand more why you make these changes?

**Comment by ceciliaavila** [comment]
Hi @limamicro,
Teams support _suggested actions_ only in 1:1 chats.
The SDK should construct a card with suggested actions whenever possible. For other types of conversations, it defaults to a hero card with options. (Per @EricDahlvang suggestion in #6086)

I'm attaching the sample used to test these changes.
[57.teams-conversation-bot.zip](https://github.com/microsoft/botbuilder-dotnet/files/11286866/57.teams-conversation-bot.zip)

**Comment by limamicro** [comment]
Hi @ceciliaavila I tested with your bot and went through the code. Your changes look good for me for the Suggested Actions part. For the Dialog in general, since I am not very familiar with the details, I asks Kavin to help with another pair of eyes. Thanks for the update!

**Review by limamicro — APPROVED** [review]
Changes related to suggested actions look good for me. Please wait for another approval from whom is familiar with the Dialog process before submitting the code. Thanks!

File: libraries/Microsoft.Bot.Builder.Dialogs/Choices/Channel.cs
Comment (limamicro) [inline libraries/Microsoft.Bot.Builder.Dialogs/Choices/Channel.cs:61]: Why there is a limitation on 3 buttons? Is it a limitation on 1:1 suggested actions?

Comment (limamicro) [reply libraries/Microsoft.Bot.Builder.Dialogs/Choices/Channel.cs:61]: @ceciliaavila, if this is a designed limitation for SA, do we need to document it?

Comment (ceciliaavila) [reply libraries/Microsoft.Bot.Builder.Dialogs/Choices/Channel.cs:61]: Teams display up to three suggested actions. I added the link to the documentation in the code.

**Review by singhk97 — APPROVED** [review]

**Review by singhk97 — APPROVED** [review]

**Review by tracyboehrer — APPROVED** [review]

**Pull request merged**
